src/main/java/org/cloudburstmc/api/util/TextFormat.java

```diff
@@ -0,0 +1,286 @@
+     */
+    LIGHT_PURPLE('d', 0xD),
+    /**
+     * Represents yellow.
+     */
+    YELLOW('e', 0xE),
+    /**
+     * Represents white.
+     */
+    WHITE('f', 0xF),
+    /**
+     * Represents minecoin gold.
+     */
+    MINECOIN_GOLD('g', 0x16),
+    /**
+     * Makes the text obfuscated.
+     */
+    OBFUSCATED('k', 0x10, true),
+    /**
+     * Makes the text bold.
+     */
+    BOLD('l', 0x11, true),
+    /**
+     * Makes a line appear through the text.
+     */
+    STRIKETHROUGH('m', 0x12, true),
+    /**
+     * Makes the text appear underlined.
+     */
+    UNDERLINE('n', 0x13, true),
+    /**
+     * Makes the text italic.
+     */
+    ITALIC('o', 0x14, true),
+    /**
+     * Resets all previous chat colors or formats.
+     */
+    RESET('r', 0x15);
+
+    /**
+     * The special character which prefixes all format codes. Use this if
+     * you need to dynamically convert format codes from your custom format.
+     */
+    public static final char ESCAPE = '\u00A7';
+
+    private static final Pattern CLEAN_PATTERN = Pattern.compile("(?i)" + ESCAPE + "[0-9A-GK-OR]");
+    private final static Map<Integer, TextFormat> BY_ID = new TreeMap<>();
+    private final static Map<Character, TextFormat> BY_CHAR = new HashMap<>();
+
+    static {
+        for (TextFormat color : values()) {
+            BY_ID.put(color.intCode, color);
+            BY_CHAR.put(color.code, color);
+        }
+    }
+
+    private final int intCode;
+    private final char code;
+    private final boolean isFormat;
+    private final String toString;
+
+    TextFormat(char code, int intCode) {
+        this(code, intCode, false);
+    }
+
+    TextFormat(char code, int intCode, boolean isFormat) {
+        this.code = code;
+        this.intCode = intCode;
+        this.isFormat = isFormat;
+        this.toString = new String(new char[]{ESCAPE, code});
+    }
+
+    /**
+     * Gets the TextFormat represented by the specified format code.
+     *
+     * @param code Code to check
+     * @return Associative {@link TextFormat} with the given code,
+     * or null if it doesn't exist
+     */
+    public static TextFormat getByChar(char code) {
+        return BY_CHAR.get(code);
+    }
+
+    /**
+     * Gets the TextFormat represented by the specified format code.
+     *
+     * @param code Code to check
+     * @return Associative {@link TextFormat} with the given code,
+     * or null if it doesn't exist
+     */
+    public static TextFormat getByChar(String code) {
+        if (code == null || code.length() <= 1) {
+            return null;
+        }
+
+        return BY_CHAR.get(code.charAt(0));
+    }
+
+    /**
+     * Cleans the given message of all format codes.
+     *
+     * @param input String to clean.
+     * @return A copy of the input string, without any formatting.
+     */
+    public static String clean(final String input) {
+        return clean(input, false);
+    }
+
+    /**
+     * Cleans the given message of all format codes.
```

Comment:
Copied over the javadoc comment from above and modified it as this method did not have one.